src/UI/Utils/useFeatures.tsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+
+const extractFeatures = (fileContent) => {
+  // Parse the fileContent and extract the value of features
+  if (!fileContent || typeof fileContent !== "string" || fileContent === "") {
+    return [];
+  }
+  const strippedStr = fileContent.replace("export", "");
+  const script = `(function() { ${strippedStr} return features; })();`;
+  return eval(script);
+};
+
+const useFeatures = (): string[] => {
+  const [features, setFeatures] = useState([]);
+
+  /* eslint-disable react-hooks/exhaustive-deps */
```

Comment:
with conditional below `extractedFeatures !== features` won't that eslint override be unnecessary